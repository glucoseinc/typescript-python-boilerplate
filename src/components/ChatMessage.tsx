import {Theme} from '@material-ui/core/styles/createMuiTheme'
import createStyles from '@material-ui/core/styles/createStyles'
import withStyles, {WithStyles} from '@material-ui/core/styles/withStyles'

import classNames from 'classnames'

import React from 'react'

import {divider, primaryText, secondaryText} from '@src/palette'
import {ChatEvent, ChatEventMessagePayload} from '@src/types'

const styles = (theme: Theme) =>
  createStyles({
    date: {
      color: secondaryText,
      fontSize: '0.6rem',
    },
    discriminator: {
      color: secondaryText,
      fontSize: '0.6rem',
    },
    meta: {},
    nickname: {
      color: primaryText,
      fontSize: '0.8rem',
    },
    root: {
      display: 'block',
      paddingTop: theme.spacing.unit,
      paddingBottom: theme.spacing.unit,
      borderBottomWidth: '1px',
      borderBottomStyle: 'solid',
      borderBottomColor: divider,
    },
  })

export interface ChatMessageProps {
  chatEvent: ChatEvent<ChatEventMessagePayload>
}

const ChatMessage: React.FunctionComponent<ChatMessageProps & WithStyles<typeof styles>> = (props) => {
  const {
    classes,
    chatEvent: {
      payload: {user, message},
      timestamp,
    },
  } = props
  const date = new Date(timestamp)

  return (
    <div className={classes.root}>
      <div className={classes.meta}>
        <span>
          <span className={classes.nickname}>{user.nickname}</span>
          <span className={classes.discriminator}>#{user.discriminator}</span>
        </span>
        <span className={classes.date}>
          {`${date.getFullYear()}-${(date.getMonth() + 1 + '').padStart(2, '0')}-` +
            `${(date.getDate() + '').padStart(2, '0')} ` +
            `${(date.getHours() + '').padStart(2, '0')}:${(date.getMinutes() + '').padStart(2, '0')}:` +
            `${(date.getSeconds() + '').padStart(2, '0')}`}
        </span>
      </div>
      <div>{message}</div>
    </div>
  )
}

export default withStyles(styles)(ChatMessage)
