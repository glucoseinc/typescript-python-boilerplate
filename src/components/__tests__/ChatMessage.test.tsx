import {shallow} from 'enzyme'
import * as React from 'react'

import {TEST_DATA} from '../__stories__/ChatMessage.stories'
import ChatMessage from '../ChatMessage'

describe('ChatMessage', () => {
  test('renders correctly', () => {
    const element = shallow(<ChatMessage {...TEST_DATA.simple} />)

    expect(element).toMatchSnapshot()
  })
})
