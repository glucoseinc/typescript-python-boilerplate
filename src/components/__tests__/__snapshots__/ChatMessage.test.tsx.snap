// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ChatMessage)
    chatEvent={
      Object {
        "localId": "69A6FEA4-84E4-473E-9689-7DC2E6F5F82C",
        "payload": Object {
          "message": "hogehgoehogehoge",
          "user": Object {
            "discriminator": "1234",
            "nickname": "test",
          },
        },
        "serverId": "69A6FEA4-84E4-473E-9689-7DC2E6F5F82C",
        "timestamp": 1546442227650,
        "type": "message",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "chatEvent": Object {
        "localId": "69A6FEA4-84E4-473E-9689-7DC2E6F5F82C",
        "payload": Object {
          "message": "hogehgoehogehoge",
          "user": Object {
            "discriminator": "1234",
            "nickname": "test",
          },
        },
        "serverId": "69A6FEA4-84E4-473E-9689-7DC2E6F5F82C",
        "timestamp": 1546442227650,
        "type": "message",
      },
      "classes": Object {
        "date": "ChatMessage-date-1",
        "discriminator": "ChatMessage-discriminator-2",
        "meta": "ChatMessage-meta-3",
        "nickname": "ChatMessage-nickname-4",
        "root": "ChatMessage-root-5",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "chatEvent": Object {
          "localId": "69A6FEA4-84E4-473E-9689-7DC2E6F5F82C",
          "payload": Object {
            "message": "hogehgoehogehoge",
            "user": Object {
              "discriminator": "1234",
              "nickname": "test",
            },
          },
          "serverId": "69A6FEA4-84E4-473E-9689-7DC2E6F5F82C",
          "timestamp": 1546442227650,
          "type": "message",
        },
        "classes": Object {
          "date": "ChatMessage-date-1",
          "discriminator": "ChatMessage-discriminator-2",
          "meta": "ChatMessage-meta-3",
          "nickname": "ChatMessage-nickname-4",
          "root": "ChatMessage-root-5",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
